R/local_registry.R

```diff
@@ -0,0 +1,80 @@
+## Generic function that computes id first, then registers it against a specified local registry 
```

Comment:
Great to see that you've abstracted the local registry. I am still curious to see how to record the context in which the registered content exists and who did the registration. A discussed earlier the recording (and registering!) of the data provenance is probably best captured in a separate package.